lib/rex/ui/text/dispatcher_shell.rb

```diff
@@ -312,7 +312,7 @@ def tab_complete_generic(fmt, str, words)
       elsif arg.kind_of?(Array)
         tabs = arg.map {|a| a.to_s}
       end
-      tabs
+      tabs.nil? ? [] : tabs
```

Comment:
It feels like the fix should live closer to the function that's returning nil 👀 

i.e. the contract for the function in question should be to always return an array, even if it's an empty array, rather than `nil` 🤔

Reply:
Additional thoughts behind that comment: There's probably other calls in the codebase that assume nil is never returned, and will crash in similar scenarios. It might be best to solve the underlying problem - rather than fix this specific side effect/bug that you're seeing 🔍

Reply:
So I have moved the code for the fix into `tab_complete_filenames(str, words)`. 

Thanks for calling this out, need to think about this more when implementing fixes. As you say, this would only have solved problems within `tab_complete_generic(fmt, str, words)`, whereas now it will solve the issue no matter where it is called.

Thanks @adfoster-r7 👍